### PR TITLE
Implement mocks with tests

### PR DESCRIPTION
**Pull request by GooGrand**

### CODE REVIEW

**Review by tornadofsoulz — DISMISSED** [review]

File: contracts/Bonding.sol [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        uint calcDecimals = sgton.calcDecimals();
+        uint secondsInYear = sgton.secondsInYear();
+        uint yearEarn = amount * calcDecimals * stakingN / stakingD;
+        return yearEarn * bondToClaimPeriod / secondsInYear / calcDecimals; 
+    }
+
+    /**
+     * View function returns timestamp when bond period vill be over
+     */
+    function bondExpiration() public view returns(uint) {
```
Comment (drinkius) [inline contracts/Bonding.sol:108]: Bond expiration normally refers to the period when a specific bond expires, so just a position of one of the clients. This function better be called bondingWindowEndTimestamp or similar

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+coverage.json
+typechain
+
+#Hardhat files
+cache
+artifacts
+types
+contracts/artifacts
+contracts/interfaces/artifacts
+contracts/libraries/artifacts
```
Comment (drinkius) [inline .gitignore:13]: Also let's add newlines at the end of each file since GitHub complains about it

File: contracts/Bonding.sol [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        (, price,,,) = _token.latestRoundData();
+    }
+
+    /**
+     * Function calculates the amount of gton out for current price without discount
+     */
+    function bondAmountOut(uint amountIn) public view returns (uint amountOut) {
+        (int256 gtonPrice, uint gtonDecimals) = tokenPriceAndDecimals(gtonAggregator);
+        (int256 tokenPrice, uint tokenDecimals) = tokenPriceAndDecimals(tokenAggregator);
+        uint tokenInUSD = amountIn * uint(tokenPrice)  / tokenDecimals;
```
Comment (drinkius) [inline contracts/Bonding.sol:126]: Extra space before /

**Review by drinkius — DISMISSED** [review]

File: contracts/BondStorage.sol [hunk 4]
```diff
@@ -0,0 +1,40 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { IBondStorage } from "./interfaces/IBondStorage.sol";
+import { AdminAccess } from "./access/AdminAccess.sol";
+import { ERC721Burnable } from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+contract BondStorage is IBondStorage,ERC721Burnable,AdminAccess {
```
Comment (drinkius) [inline contracts/BondStorage.sol:9]: Missing spaces after commas

**Comment by alexpgc** [comment]
@drinkius is this pr makes sense to be merged?

**Review by drinkius — DISMISSED** [review]

**Comment by drinkius** [comment]
@alexpgc yes, approved it, upon your approve we can merge this

**Review by drinkius — DISMISSED** [review]

**Review by drinkius — DISMISSED** [review]

**Review by drinkius — APPROVED** [review]

**Review by alexpgc — APPROVED** [review]

**Review by drinkius — APPROVED** [review]

**Pull request merged**